src/Models/HydrostaticFreeSurfaceModels/implicit_free_surface.jl

```diff
@@ -147,23 +145,16 @@ function implicit_free_surface_step!(free_surface::ImplicitFreeSurface, model, 
 
     fill_halo_regions!(η)
     
-    return prognostic_field_events
-end
-
-function wait_velocity_event(arch, prognostic_field_events)
-    velocity_events = prognostic_field_events[1]
-
-    # Wait for predictor velocity update step to complete.
-    wait(device(arch), MultiEvent(velocity_events))
-
-    return MultiEvent(prognostic_field_events[2])
+    return nothing
 end
 
 function local_compute_integrated_volume_flux!(∫ᶻQ, velocities, arch)
     
-    masking_events = Tuple(mask_immersed_field!(q, blocking=false) for q in velocities)
-    wait(device(arch), MultiEvent(masking_events))
-
+    
+    for u in velocities
+        mask_immersed_field!(u)
+    end
```

Comment:
```suggestion
    for q in velocities
        mask_immersed_field!(q)
    end
```

trivial, but hey... (there's velocities.u so it could confuse)

there's also `mask_immersed_velocities!`. I don't know why though.